bee-network/bee-autopeering/src/request.rs

```diff
@@ -0,0 +1,202 @@
+        let timestamp = verif_req.timestamp();
+
+        let request_hash = msg_hash(
+            MessageType::VerificationRequest,
+            &verif_req.to_protobuf().expect("error encoding verification request"),
+        );
+
+        let value = RequestValue {
+            request_hash,
+            expiration_time: timestamp + REQUEST_EXPIRATION_SECS,
+            response_tx,
+        };
+
+        let _ = self.open_requests.insert(key, value);
+
+        verif_req
+    }
+
+    pub(crate) fn new_discovery_request(
+        &mut self,
+        peer_id: PeerId,
+        response_tx: Option<ResponseTx>,
+    ) -> DiscoveryRequest {
+        let key = RequestKey {
+            peer_id,
+            request_id: TypeId::of::<DiscoveryRequest>(),
+        };
+
+        let disc_req = DiscoveryRequest::new();
+        let timestamp = disc_req.timestamp();
+
+        let request_hash = msg_hash(
+            MessageType::DiscoveryRequest,
+            &disc_req.to_protobuf().expect("error encoding discovery request"),
+        );
+
+        let value = RequestValue {
+            request_hash,
+            expiration_time: timestamp + REQUEST_EXPIRATION_SECS,
+            response_tx,
+        };
+
+        let _ = self.open_requests.insert(key, value);
+
+        disc_req
+    }
+
+    pub(crate) fn new_peering_request(
+        &mut self,
+        peer_id: PeerId,
+        response_tx: Option<ResponseTx>,
+        local: &Local,
+    ) -> PeeringRequest {
+        let key = RequestKey {
+            peer_id,
+            request_id: TypeId::of::<PeeringRequest>(),
+        };
+
+        let peer_req = PeeringRequest::new(local.public_salt().expect("missing public salt"));
+
+        let timestamp = peer_req.timestamp();
+
+        let request_hash = msg_hash(
+            MessageType::PeeringRequest,
+            &peer_req.to_protobuf().expect("error encoding peering request"),
+        );
+
+        let value = RequestValue {
+            request_hash,
+            expiration_time: timestamp + REQUEST_EXPIRATION_SECS,
+            response_tx,
+        };
+
+        let _ = self.open_requests.insert(key, value);
+
+        peer_req
+    }
+
+    pub(crate) fn pull<R: Request + 'static>(&mut self, peer_id: &PeerId) -> Option<RequestValue> {
+        let key = RequestKey {
+            peer_id: *peer_id,
+            request_id: TypeId::of::<R>(),
+        };
+
+        self.open_requests.remove(&key)
+    }
+}
+
+pub(crate) fn is_expired(timestamp: Timestamp) -> bool {
+    time::since(timestamp).map_or(false, |ts| ts >= REQUEST_EXPIRATION_SECS)
+}
+
+pub(crate) fn remove_expired_requests_repeat() -> Repeat<RequestManager> {
+    Box::new(|mngr: &RequestManager| {
+        // Retain only those that aren't expired yet, remove all others.
+        mngr.write()
+            .open_requests
+            .retain(|_, v| v.expiration_time > time::unix_now_secs());
```

Comment:
It's a bit weird that we just created an `is_expired` function but don't use it here

Reply:
oh yea, I agree. Fixing this....

Reply:
Done

Reply:
is this doing a syscall for every request? would it be too bad to store `time::unix_now_secs()` beforehand and use it for all the request?

Reply:
Done